mayur/cart/inventory.rb

```diff
@@ -0,0 +1,29 @@
+class Inventory
+  attr_accessor  :products, :codes
+  def initialize(products = [], codes = {})
+    @products = products
+    @codes = codes
+  end
+
+  def delete_item(name = nil)
+    index = @products.find_index { |p| p[:name] == name }
+    return nil if index.nil?
+    @products.delete_at(index)
+  end
+
+  def add_item(name = nil, price = nil, quantity = nil)
+    return nil if (name.nil? || quantity.nil?) 
+    index = @products.find_index { |p| p[:name] == name }
```

Comment:
May we avoid using index here? We can do this using few other methods like `map`

Reply:
Map does not do the job